client/pages/home.jsx

```diff
@@ -13,19 +13,46 @@ export default class Home extends React.Component {
     super(props);
     this.state = {
       entries: [],
-      seeModal: false,
+      seeLightbox: false,
+      seeDelete: false,
+      deleting: null,
       lightboxImage: null
     };
-    this.showModal = this.showModal.bind(this);
-    this.hideModal = this.hideModal.bind(this);
+    this.showLightbox = this.showLightbox.bind(this);
+    this.hideLightbox = this.hideLightbox.bind(this);
+    this.showDelete = this.showDelete.bind(this);
+    this.hideDelete = this.hideDelete.bind(this);
+    this.deleteEntry = this.deleteEntry.bind(this);
   }
 
-  showModal(event) {
-    this.setState({ seeModal: true, lightboxImage: event.target.getAttribute('src') });
+  showLightbox(event) {
+    this.setState({ seeLightbox: true, lightboxImage: event.target.getAttribute('src'), seeDelete: false });
   }
 
-  hideModal() {
-    this.setState({ seeModal: false });
+  showDelete(event) {
+    this.setState({ seeDelete: true, deleting: event.target.getAttribute('id') });
+  }
+
+  hideLightbox() {
+    this.setState({ seeLightbox: false });
+  }
+
+  hideDelete() {
+    this.setState({ seeDelete: false });
+  }
+
+  deleteEntry() {
+    const { deleting, entries } = this.state;
+    const numDeleting = Number(deleting);
+    const stateCopy = entries.filter(entry => entry.entryId !== numDeleting);
+    const req = {
+      method: 'DELETE',
+      headers: { 'Content-Type': 'application/json' }
+    };
+    fetch(`/api/entries/${deleting}`, req)
+      .then(res => res.json());
+    this.setState({ entries: stateCopy, deleting: null });
+    this.hideDelete();
```

Comment:
To make sure we don't update our state prematurely, these method calls of lines 54 and 55 should be moved into the fetch's `.then()` callback